ethcore/src/engines/authority_round/mod.rs

```diff
@@ -1721,12 +1725,68 @@ mod tests {
 
 		engine.set_signer(Box::new((tap, addr2, "2".into())));
 		if let Seal::Regular(seal) = engine.generate_seal(&b2, &genesis_header) {
+			// FIXME: This branch is unreachable because the call to `generate_seal` above always
```

Comment:
Nice catch

I think it is better to have `if-else` or something similar just to be explicit for what is expected behavior. 

```rustc
if let Seal::Regular(seal) = engine.generate_seal(&b1, &genesis_header) {
...
} else {
   assert_eq!(1, 0, "Seal::Regular is expected here")
}
assert_eq!(engine.generate_seal(&b2, &genesis_header), Seal::None), "Engine is not stepped; thus no blocks accepted at the same time")
```

However, I'm not convinced if we need the second block for this test perhaps just remove it.

/cc @debris

Reply:
Yes, it depends on what the expected behaviour was supposed to be. Wouldn't the test will always fail in the `else` branch that you proposed? The last assertion after `if-else` is unreachable.

Reply:
yes, if we would two identical ones without stepping the engine (as it is currently) then it would fail.

But I'm more in favor of just to test one block in the test `generates_seal_and_does_not_double_propose` or simply remove it in favor of `your added test "generates_seal_iff_sealer_is_set"`. It seems redundant now..

Reply:
I agree, `generates_seal_and_does_not_double_propose` does seem a bit redundant in view of having the same test being part of `generates_seal_iff_sealer_is_set`. I'll keep it for now and remove only the test of the second block.